tabular/src/autogluon/tabular/models/tabular_nn/tabular_nn_model.py

```diff
@@ -343,7 +343,12 @@ def train_net(self, train_dataset, params, val_dataset=None, initialize=True, se
             if val_dataset is not None:
                 val_metric = self.score(X=val_dataset, y=y_val, metric=self.stopping_metric)
                 if (val_metric >= best_val_metric) or (e == 0):
-                    if not np.isnan(val_metric):
+                    if np.isnan(val_metric):
+                        if e == 0:
+                            raise RuntimeError("NaNs encountered in TabularNeuralNetModel training. Features/labels may be improperly formatted or NN weights may have diverged.")
+                        else:
+                            break
```

Comment:
add warning here, this should be visible to us when this happens, as it something we don't want to occur.